frontend/src/components/S3ArtifactLink.tsx

```diff
@@ -0,0 +1,31 @@
+import * as React from 'react';
+import { StoragePath, StorageService } from '../lib/WorkflowParser';
+import { S3Artifact } from '../../third_party/argo-ui/argo_template';
+
+const artifactApiUri = ({ source, bucket, key }: StoragePath) =>
+  'artifacts/get' +
+  `?source=${source}&bucket=${bucket}&key=${encodeURIComponent(key)}`;
+
+/**
+ * A component that renders an artifact link.
+ */
+const S3ArtifactLink: React.FC<S3Artifact> = (
```

Comment:
May I ask for a really simple test for this component?

Reply:
added 3 simple test case:
- invalid artifact (empty bucket or key)
- s3 artifact
- minio artifact

Reply:
Thanks! LGTM